### PR TITLE
Fix issue when upgrading logback to 1.1.5 beyond

### PR DESCRIPTION
**Pull request by xiaodong-xie**
Fix https://github.com/dropwizard/dropwizard/issues/1478

### CODE REVIEW

File: dropwizard-logging/src/main/java/io/dropwizard/logging/FileAppenderFactory.java [hunk 1]
```diff
@@ -199,7 +199,6 @@ public boolean isMaxFileSizeSettingSpecified() {
         appender.setPrudent(false);
         appender.addFilter(levelFilterFactory.build(threshold));
         getFilterFactories().stream().forEach(f -> appender.addFilter(f.build()));
-        appender.stop();
```
Comment (xiaodong-xie) [inline dropwizard-logging/src/main/java/io/dropwizard/logging/FileAppenderFactory.java]: If this line was not removed, test `DefaultLoggingFactoryTest::testConfigure` will fail

**Comment by xiaodong-xie** [comment]
If we only upgrade logback to 1.1.5, build will pass: https://travis-ci.org/dropwizard/dropwizard/builds/116483726

But logback 1.1.6 introduced some other more strict stuff: (http://logback.qos.ch/codes.html#earlier_fa_collision)
```
File/FileNamePattern option has the same value "..." as that given for appender [...] defined earlier.

If a FileAppender/RollingFileAppender defined earlier has the same File option as the current appender, then those two appenders are in collision as FileAppender instances cannot share the same output target. To prevent loss of data, the current appender will not start. Make sure that each appender has a unique File option.

By analogy the same restriction applies to the FileNamePattern option of RollingFileAppender. Make sure that each RollingFileAppender has a unique FileNamePattern option
```

So if we want to upgrade 1.1.6, we need to tweak `yaml/logging_advanced.yml` file...

**Comment by arteam** [comment]
The fix looks good, thanks you very much for contributing! 

I think we should not hesitate to upgrade to 1.1.6. As I see it, the release brings several improvements (e.g. disabling packaging data) and many bug fixes, which should beneficial for Dropwizard users. Judging by the rate of releases, the logback development community actively fixes other discovered bugs. I think we can rely on it and expect that potential bugs, with which Dropwizard users may encounter, will be fixed.

**Comment by xiaodong-xie** [comment]
@arteam, thanks for merging... :)

**Pull request merged**